src/forms/validators/blueprintNameValidator.js

```diff
@@ -0,0 +1,23 @@
+/* eslint-disable react/display-name */
+import React from "react";
+import { FormattedMessage } from "react-intl";
+import { selectAllBlueprintNames } from "../../slices/blueprintsSlice";
+import store from "../../store";
+
+const blueprintNameValidator = () => (value) => {
+  const state = store.getState();
+  const blueprintNames = selectAllBlueprintNames(state);
```

Comment:
Can you move this into the `BlueprintWizard` component and pass it into the validator. Currently, every key press will call the store state.

Also, you can use the `useSelector` hook instead of importing the store. 

```
const blueprintNames = useSelector((state) => selectAllBlueprintNames(state));
```

Reply:
Good idea, didn't realize you could pass data into DDF validators!

Reply:
Ya I'm not sure if you pass it directly to the validator or to the form state which you access from the validator. Docs aren't the clearest so you might need to play around :D